### PR TITLE
added net4.8 support

### PR DESCRIPTION
**Pull request by ankitdas13**

### CODE REVIEW

File: Razorpay.nuspec [hunk 1]
```diff
@@ -22,5 +22,6 @@
     <file src="net40/bin/Release/Razorpay.dll" target="lib/net40/Razorpay.dll" />
     <file src="net45/bin/Release/Razorpay.dll" target="lib/net45/Razorpay.dll" />
     <file src="net47/bin/Release/Razorpay.dll" target="lib/net47/Razorpay.dll" />
+	  <file src="net48/bin/Release/Razorpay.dll" target="lib/net48/Razorpay.dll" />
```
Comment (shwatang) [inline Razorpay.nuspec:25]: Indentation fix.

File: net48/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+// General Information about an assembly is controlled through the following
+// set of attributes. Change these attribute values to modify the information
+// associated with an assembly.
+[assembly: AssemblyTitle("RazorpayClient48")]
+[assembly: AssemblyDescription("")]
+[assembly: AssemblyConfiguration("")]
+[assembly: AssemblyCompany("")]
+[assembly: AssemblyProduct("RazorpayClient48")]
+[assembly: AssemblyCopyright("Copyright ©  2024")]
```
Comment (shwatang) [inline net48/Properties/AssemblyInfo.cs:13]:
```suggestion
[assembly: AssemblyTitle("Razorpay")]
[assembly: AssemblyDescription("Razorpay SDK for .Net")]
[assembly: AssemblyConfiguration("")]
[assembly: AssemblyCompany("Razorpay Software Private Limited")]
[assembly: AssemblyProduct("Razorpay.API")]
[assembly: AssemblyCopyright("Copyright © 2015 - 2024")]
```
Should't it be this? Correct me if mistaken.

File: net48/Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+// Version information for an assembly consists of the following four values:
+//
+//      Major Version
+//      Minor Version
+//      Build Number
+//      Revision
+//
+// You can specify all the values or you can default the Build and Revision Numbers
+// by using the '*' as shown below:
+// [assembly: AssemblyVersion("1.0.*")]
```
Comment (shwatang) [inline net48/Properties/AssemblyInfo.cs:34]: Can we remove this?

File: net48/packages.config [hunk 4]
```diff
@@ -0,0 +1,5 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="Newtonsoft.Json" version="13.0.3" targetFramework="net48" />
+  <package id="Portable.BouncyCastle" version="1.8.0" targetFramework="net48" />
```
Comment (shwatang) [inline net48/packages.config:4]: why changed it from 1.9.0 version?

File: net48/RazorpayClient48.csproj [hunk 5]
```diff
@@ -0,0 +1,199 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (shwatang) [inline net48/RazorpayClient48.csproj:2]: why we not adding DefaultTargets key?

File: net48/RazorpayClient48.csproj [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
+    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
+    <ProjectGuid>{35244CCA-2467-4D21-AA86-A37791358950}</ProjectGuid>
+    <OutputType>Library</OutputType>
+    <AppDesignerFolder>Properties</AppDesignerFolder>
+    <RootNamespace>Razorpay.Api</RootNamespace>
+    <AssemblyName>Razorpay</AssemblyName>
+    <TargetFrameworkVersion>v4.8</TargetFrameworkVersion>
+    <FileAlignment>512</FileAlignment>
+    <Deterministic>true</Deterministic>
```
Comment (shwatang) [inline net48/RazorpayClient48.csproj:14]: What this does?

File: net48/RazorpayClient48.csproj [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+  </PropertyGroup>
+  <PropertyGroup>
+    <StartupObject />
+  </PropertyGroup>
+  <ItemGroup>
+    <Reference Include="crypto, Version=1.8.0.0, Culture=neutral, PublicKeyToken=0e99375e54769942, processorArchitecture=MSIL">
+      <HintPath>packages\Portable.BouncyCastle.1.8.0\lib\net46\crypto.dll</HintPath>
+    </Reference>
+    <Reference Include="Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
+      <HintPath>packages\Newtonsoft.Json.13.0.3\lib\net45\Newtonsoft.Json.dll</HintPath>
```
Comment (shwatang) [inline net48/RazorpayClient48.csproj:41]: Please once check the HintPath and confirm if its correct.

File: net48/RazorpayClient48.csproj [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+    <Reference Include="System" />
+    <Reference Include="System.IO.FileSystem, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
+      <HintPath>packages\System.IO.FileSystem.4.0.0\lib\net46\System.IO.FileSystem.dll</HintPath>
+      <Private>True</Private>
+      <Private>True</Private>
+    </Reference>
+    <Reference Include="System.IO.FileSystem.Primitives, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
+      <HintPath>packages\System.IO.FileSystem.Primitives.4.0.0\lib\net46\System.IO.FileSystem.Primitives.dll</HintPath>
+      <Private>True</Private>
+      <Private>True</Private>
```
Comment (shwatang) [inline net48/RazorpayClient48.csproj:52]: Why including this as part this PR?

**Review by shwatang — COMMENTED** [review]
Please address the comments. Also, please do confirm if this change has been tested thoroughly.

File: net48/Properties/AssemblyInfo.cs (hunk 2 above)
Comment (ankitdas13) [reply net48/Properties/AssemblyInfo.cs]: updated 

https://github.com/razorpay/razorpay-dot-net/blob/10148cd90cfb84ff83d6f4e8643488f5fe22eef5/net48/Properties/AssemblyInfo.cs#L8

File: net48/packages.config (hunk 4 above)
Comment (ankitdas13) [reply net48/packages.config]: updated 

https://github.com/razorpay/razorpay-dot-net/blob/10148cd90cfb84ff83d6f4e8643488f5fe22eef5/net48/packages.config#L4

File: net48/RazorpayClient48.csproj (hunk 5 above)
Comment (ankitdas13) [reply net48/RazorpayClient48.csproj]: added 

https://github.com/razorpay/razorpay-dot-net/blob/10148cd90cfb84ff83d6f4e8643488f5fe22eef5/net48/RazorpayClient48.csproj#L2

File: net48/RazorpayClient48.csproj (hunk 6 above)
Comment (ankitdas13) [reply net48/RazorpayClient48.csproj]: removed

File: net48/RazorpayClient48.csproj [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+    <OutputPath>bin\Release\</OutputPath>
+    <DefineConstants>TRACE</DefineConstants>
+    <ErrorReport>prompt</ErrorReport>
+    <WarningLevel>4</WarningLevel>
+  </PropertyGroup>
+  <PropertyGroup>
+    <StartupObject />
+  </PropertyGroup>
+  <ItemGroup>
+    <Reference Include="crypto, Version=1.8.0.0, Culture=neutral, PublicKeyToken=0e99375e54769942, processorArchitecture=MSIL">
```
[the remark this answers is not in the document]
Comment (ankitdas13) [reply net48/RazorpayClient48.csproj]: updated

https://github.com/razorpay/razorpay-dot-net/blob/10148cd90cfb84ff83d6f4e8643488f5fe22eef5/net48/RazorpayClient48.csproj#L36

File: net48/RazorpayClient48.csproj (hunk 7 above)
Comment (ankitdas13) [reply net48/RazorpayClient48.csproj]: its correct

Screenshot 2024-06-20 153132

File: net48/RazorpayClient48.csproj (hunk 8 above)
Comment (ankitdas13) [reply net48/RazorpayClient48.csproj:51]: to use `System.Net.HttpWebRequest` namespace in net4.8 we need to add reference.

**Comment by ankitdas13** [comment]
> Please address the comments. Also, please do confirm if this change has been tested thoroughly.

I addressed these comments and made changes, as well as testing it

**Review by shwatang — APPROVED** [review]
LGTM

**Pull request merged**